### ISSUE TITLE
Facebook dosen't generate a card

### ISSUE THREAD

**Issue by paladox** [octfx/wiki-seo#10]
Hi, we have a bug report here https://phabricator.miraheze.org/T4285#103524 that facebook is not showing it as a card, rather just a plain url.

**Comment by paladox** [comment]
Facebook also says "The following required properties are missing: og:type, og:description". Also got "The specified 'fb:app:id' property is not a valid app ID.".

**Comment by octfx** [comment]
Thanks for reporting the issue, I'll have a look sometime next week.

**Comment by octfx** [comment]
The issue seems to be related to some new features introduced to the extension.  

The Extension adds some base SEO tags like `article:modified_time`, `og:locale`, `og:title` by accessing data from the database.  
One feature that was added is, that the configured site logo is used in tags like `og:image` if no image is set either in the page itself through the parser function or a default image set in LocalSettings.  
This would explain why the logo is used instead of the first image in the page.  

I'll add a flag to disable this behaviour.


As for the `fb:app:id` this will be changed/added to `fb:admins`.  
`og:type` and `og:description` is missing as this is not set automatically by the extension.

**Comment by paladox** [comment]
Is there a way to get og:description set automatically to the first line of the article?

Also facebook dosen't show anything (it just shows a link). See https://www.facebook.com/LaNonciclopedia/posts/3208473025829645

**Comment by octfx** [comment]
In theory WikiSEO uses the description provided by the Description2 extension. I'll have to investitage as to why this is not happening in the provided case.

Putting the link from the post into facebooks [sharing debugger](https://developers.facebook.com/tools/debug/?q=https%3A%2F%2Fnonciclopedia.org%2Fwiki%2FRoberto_Burioni) show a card with image and text, so the tags should work in theory.

**Comment by paladox** [comment]
bump, any update? :)

**Comment by octfx** [comment]
I haven't forgotten, but time is short at the moment. 😕
But my schedule will clear up soon, I think at the end of the month I can take care of it!

**Comment by paladox** [comment]
thanks!

**Comment by octfx** [comment]
I've updated the extension to pick up the description provided by Description2.  
Two new flags were added to further refine the behavior of the extension.  
`$wgWikiSeoDisableLogoFallbackImage = true;` - The wiki logo will no longer be used if no image was provided (default is `false`).  
`$wgFacebookAdmins = 'ID1,ID2,...'`; - Adds ids to the `fb:admins` meta tag.

While the changes are in master, there is currently no tagged version. @paladox
